pyroute2/dhcp/dhcp4socket.py

```diff
@@ -20,22 +21,40 @@
 
 UDP_HEADER_SIZE = 8
 IPV4_HEADER_SIZE = 20
+SKF_AD_OFF = -0x1000
+SKF_AD_VLAN_TAG_PRESENT = 48
 
 
 def listen_udp_port(port: int = 68) -> list[list[int]]:
-    # pre-scripted BPF code that matches UDP port
+    '''BPF filter that matches Ethernet + IPv4 + UDP on the given port.
+
+    Packets tagged on a vlan are also dropped, see
+    https://lore.kernel.org/netdev/51FB6A9D.2050002@redhat.com/T/
+    '''
     bpf_code = [
```

Comment:
[not a request for changes] I wonder, if we add some basic BPF compiler one day :)))

Reply:
While working on the filter i was thinking exactly the same thing, i hope to come across a use case that will require it one day :) for now having constants instead of raw int values is a first step, and you already wrote `compile()` in bpf.py which is arguably a very basic BPF compiler !